LibreNMS/DB/SyncsModels.php

```diff
@@ -45,7 +45,10 @@ protected function syncModels($device, $relationship, $models): Collection
 
         foreach ($existing as $exist_key => $exist_value) {
             if ($models->offsetExists($exist_key)) {
-                // update
+                // update or restore soft-deleted model
+                if ($exist_value->trashed()) {
+                    $exist_value->restore();
```

Comment:
Does anything bad happen if you just call restore on a non-trashed model?

Reply:
Oh, you actually need to check the the model uses soft deletes instead of if it is trashed.

Reply:
Yep